frontend/catalyst/jax_tracer.py

```diff
@@ -962,24 +955,30 @@ def trace_quantum_measurements(
     the quantum measurement outputs, leave other outputs as-is.
 
     Args:
+        shots_obj: Shots object containing shots information (total_shots, shot_vector, num_copies).
+                  The total_shots can be an int or JAX tracer for dynamic shots.
         device (QubitDevice): PennyLane quantum device to use for quantum measurements.
         qrp (QRegPromise): Quantum register tracer with cached qubits
         outputs (List of quantum function results): List of qnode output JAX tracers to process.
         out_tree (PyTreeDef): PyTree-shape of the outputs.
-        quantum_tape: PennyLane quantum tape.
 
     Returns:
         out_classical_tracers: modified list of JAX classical qnode ouput tracers.
         out_tree: modified PyTree-shape of the qnode output.
     """
-    shots = get_device_shots(device)
+    # Extract total shots and detect if dynamic
+    shots = shots_obj.total_shots if shots_obj else None
     out_classical_tracers = []
 
     for i, output in enumerate(outputs):
         if isinstance(output, MeasurementProcess):
 
             # Check if the measurement is supported shot-vector where num_of_total_copies > 1
-            if device.shots.num_copies > 1 and not isinstance(output, qml.measurements.SampleMP):
+            if (
+                shots_obj
+                and shots_obj.num_copies > 1
```

Comment:
```suggestion
                shots_obj.num_copies > 1
```
If the `shots_obj` is falsy, then `num_copies == 0`.